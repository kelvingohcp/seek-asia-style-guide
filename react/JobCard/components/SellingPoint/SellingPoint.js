import React from 'react';
import PropTypes from 'prop-types';
import styles from './SellingPoint.less';
import Text from '../../../Text/Text';

const SellingPoint = ({ sellingPoints, showSellingPoint }) => {
  if (!showSellingPoint || !sellingPoints) {
    return null;
  }

  return (
    <ul className={styles.sellingPoints}>
      {sellingPoints.map((sellingPoint, i) => (
        <li key={i} className={styles.sellingPoint}>
          <Text
            whispering
            baseline={false}
            className={styles.text}>
            {sellingPoint}
          </Text>
        </li>
      ))}
    </ul>
  );
};

SellingPoint.propTypes = {
  sellingPoints: PropTypes.arrayOf(PropTypes.string),
  showSellingPoint: PropTypes.bool
};

export default SellingPoint;
