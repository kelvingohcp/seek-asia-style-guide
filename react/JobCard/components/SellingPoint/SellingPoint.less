@import (reference) '~seek-asia-style-guide/theme';
@import (reference) '../../JobCard';

.sellingPoints {
  .rawText();
  @media @aboveMobile {
    padding-top: @grid-base * 3;
  }
  padding-top: @grid-base * 2;
  color: @saGrey3;
  list-style: none;
}

.text {
  color: @saGrey1;
  text-overflow: ellipsis;
  overflow: hidden;
  padding-bottom: @grid-base;
}

.sellingPoint {
  display: flex;

  &::before {
    content: "\2022";
    width: @grid-base * 4;
    text-align: center;
    margin-right: @grid-base * 2;
    display: inline-block;
    flex: 0 0 auto;
    line-height: @grid-base * 3;
  }
}
