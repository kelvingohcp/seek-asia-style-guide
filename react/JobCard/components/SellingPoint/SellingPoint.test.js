import React from 'react';
import { shallow } from 'enzyme';
import SellingPoint from './SellingPoint';

const sellingPoints = [
  'sellingPoints 1',
  'sellingPoints 2',
  'sellingPoints 3'
];

describe('SellingPoint', () => {
  it('should render correctly', () => {
    expect(shallow(<SellingPoint sellingPoints={sellingPoints} isSplitView showSellingPoint enableBrandedAd />)).toMatchSnapshot();
  });

  it('should not render selling points when showSellingPoint is false', () => {
    expect(shallow(<SellingPoint sellingPoints={sellingPoints} isSplitView showSellingPoint={false} enableBrandedAd />)).toMatchSnapshot();
  });

  it('should not render selling points when no sellingPoints', () => {
    expect(shallow(<SellingPoint sellingPoints={null} isSplitView showSellingPoint enableBrandedAd />)).toMatchSnapshot();
  });
});
