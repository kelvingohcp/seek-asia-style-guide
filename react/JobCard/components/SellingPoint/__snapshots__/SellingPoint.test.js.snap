// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellingPoint should not render selling points when no sellingPoints 1`] = `""`;

exports[`SellingPoint should not render selling points when showSellingPoint is false 1`] = `""`;

exports[`SellingPoint should render correctly 1`] = `
<ul
  className="sellingPoints"
>
  <li
    className="sellingPoint"
    key="0"
  >
    <Text
      baseline={false}
      className="text"
      whispering={true}
    >
      sellingPoints 1
    </Text>
  </li>
  <li
    className="sellingPoint"
    key="1"
  >
    <Text
      baseline={false}
      className="text"
      whispering={true}
    >
      sellingPoints 2
    </Text>
  </li>
  <li
    className="sellingPoint"
    key="2"
  >
    <Text
      baseline={false}
      className="text"
      whispering={true}
    >
      sellingPoints 3
    </Text>
  </li>
</ul>
`;
